godot-macros/src/lib.rs

```diff
@@ -526,126 +526,150 @@ pub fn godot_api(_meta: TokenStream, input: TokenStream) -> TokenStream {
+/// type.
 ///
-/// This also requires deriving `GodotConvert`.
+/// The types you can use this with currently are:
+/// - `GString`
+/// - `i8`, `i16`, `i32`, `i64`
+/// - `u8`, `u16`, `u32`
 ///
-/// Currently has some tight requirements which are expected to be softened as implementation expands:
-/// - Only works for enums, structs aren't supported by this derive macro at the moment.
-/// - The enum must have an explicit `#[repr(u*/i*)]` type.
-///     - This will likely stay this way, since `isize`, the default repr type, is not a concept in Godot.
-/// - The enum variants must not have any fields - currently only unit variants are supported.
-/// - The enum variants must have explicit discriminants, that is, e.g. `A = 2`, not just `A`
+/// When using one of the integer types, each variant of the enum will be converted into its discriminant.
 ///
-/// # Example
+/// ### Examples
 ///
 /// ```no_run
-/// # use godot::prelude::*;
-/// #[derive(Var, GodotConvert)]
-/// #[repr(i32)]
-/// # #[derive(Eq, PartialEq, Debug)]
+/// use godot::prelude::*;
+/// #[derive(GodotConvert)]
+/// #[godot(via = GString)]
 /// enum MyEnum {
-///     A = 0,
-///     B = 1,
+///     A,
+///     B,
+///     C,
 /// }
 ///
-/// #[derive(GodotClass)]
-/// #[class(no_init)] // No Godot default constructor.
-/// struct MyClass {
-///     #[var]
-///     foo: MyEnum,
+/// assert_eq!(MyEnum::A.to_godot(), GString::from("A"));
+/// assert_eq!(MyEnum::B.to_godot(), GString::from("B"));
+/// assert_eq!(MyEnum::C.to_godot(), GString::from("C"));
+/// ```
+///
+/// ```no_run
+/// use godot::prelude::*;
+/// #[derive(GodotConvert)]
+/// #[godot(via = i64)]
+/// enum MyEnum {
+///     A,
+///     B,
+///     C,
 /// }
 ///
-/// fn main() {
-///     let mut class = MyClass { foo: MyEnum::B };
-///     assert_eq!(class.get_foo(), MyEnum::B as i32);
+/// assert_eq!(MyEnum::A.to_godot(), 0);
+/// assert_eq!(MyEnum::B.to_godot(), 1);
+/// assert_eq!(MyEnum::C.to_godot(), 2);
+/// ```
+///
+/// Explicit discriminants are used for integers:
 ///
-///     class.set_foo(MyEnum::A as i32);
-///     assert_eq!(class.foo, MyEnum::A);
+/// ```no_run
+/// use godot::prelude::*;
+/// #[derive(GodotConvert)]
+/// #[godot(via = u8)]
+/// enum MyEnum {
+///     A,
+///     B = 10,
+///     C,
 /// }
+///
+/// assert_eq!(MyEnum::A.to_godot(), 0);
+/// assert_eq!(MyEnum::B.to_godot(), 10);
+/// assert_eq!(MyEnum::C.to_godot(), 11);
 /// ```
-#[proc_macro_derive(Var)]
-pub fn derive_property(input: TokenStream) -> TokenStream {
+#[proc_macro_derive(GodotConvert, attributes(godot))]
+pub fn derive_godot_convert(input: TokenStream) -> TokenStream {
+    translate(input, derive::derive_godot_convert)
+}
+
+/// Derive macro for [`Var`](../register/property/trait.Var.html) on enums.
+///
+/// This expects a derived [`GodotConvert`](../builtin/meta/trait.GodotConvert.html) implementation, using a manual
+/// implementation of `GodotConvert` may lead to incorrect values being displayed in Godot.
```

Comment:
Have you encountered this? I think it's good advice, but I wonder what concretely you thought of here?

Reply:
if you did say
```rs
#[derive(Var, Export)]
#[godot(via = i64)]
pub enum MyEnum {
  A,
  B,
}

impl GodotConvert for MyEnum {
  type Via = i64;
}

impl FromGodot for MyEnum {
  fn try_from_Godot(via: Self::Via) -> Result<..> {
    match via {
      1 => Ok(Self::A),
      0 => Ok(Self::B),
      _ => Err(..)
    }
  }
}
// And corresponding ToGodot
```
Then the `Var` and `Export` impl would map `A` and `B` to each other, since the assumption is that `A = 0` and `B = 1` since that's their rust discriminants.

Reply:
Thanks for elaborating!

Your suggestion sounds good, but would require that any `ToGodot` implementation by the user here also support `Display` -- and that `Display` then match the property-hint format. It's mixing responsibilities a bit. Either a dedicated method/trait, or conversion to `Variant` and using its `Display` might be an option...

But for now, the limitation is OK I think, especially since it's documented.